src/common/imageio.c

```diff
@@ -78,6 +78,38 @@
 #include "lua/image.h"
 #endif
 
+static const gchar *_supported_raw[]
+    = { "3fr", "ari", "arw", "bay", "cr2", "cr3", "crw", "dc2", "dcr", "dng", "erf", "fff",
+        "ia",  "iiq", "k25", "kc2", "kdc", "mdc", "mef", "mos", "mrw", "nef", "nrw", "orf",
+        "pef", "raf", "raw", "rw2", "rwl", "sr2", "srf", "srw", "sti", "x3f" };
+/* commented out but good to keep around as a reference
+static const gchar *_supported_ldr[]
+    = { "bmp",  "bmq", "cap", "cine", "cs1", "dcm", "gif", "gpr", "j2c", "j2k", "jng", "jp2", "jpc", "jpeg", "jpg",
+        "miff", "mng", "ori", "pbm",  "pfm", "pgm", "png", "pnm", "ppm", "pxn", "qtk", "rdc", "tif", "tiff" };*/
+static const gchar *_supported_hdr[] = { "avif", "exr", "hdr", "heic", "heif", "hif", "pfm" };
+
+// get the type of image from its extension
+dt_image_flags_t dt_imageio_get_type_from_extension(const char *extension)
+{
+  const char *ext = g_str_has_prefix(extension, ".") ? extension + 1 : extension;
+  for(const char **i = _supported_raw; *i != NULL; i++)
```

Comment:
tbh, it's a copy-paste of the code in `dt_supported_image` from src/common/darktable.c :)

Reply:
You copy-pasted it wrong, dt_supported_extensions array is NULL-terminated, while yours _supported_* are not, so you have here a completely undefined behaviour (crash, wrongly detected types, etc.). I will push a fix.

Reply:
Which won't fix already wrongly set types (was this tested at all?)

Reply:
good catch. and thanks for the fix !
Yes this has been tested (it's in master for more than 3 weeks now...) but it seems that somehow our compiler was sneaky enough to hide this bug...

Reply:
No, I understand that it didn’t crash on Linux or Windows, because NULL value happen to be present somewhere close-by in memory. My point was that for example if compiler put those 3 arrays in the memory as they are declared, then this first loop will match all extensions and this function will always return DT_IMAGE_RAW. I was asking if anyone tested it actually working as advertised, not whether it crashed. I suspect nobody actually uses this functionality, especially since it’s unreliable by definition, so being present in master for several weeks doesn’t mean much in this context. Anyway, doesn’t matter much anymore.